x/ecocredit/basket/msg_create.go

```diff
@@ -0,0 +1,48 @@
+package basket
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var (
+	_ legacytx.LegacyMsg = &MsgCreate{}
+)
+
+///
+/// MsgCreate sdk.Msg interface
+///
+
+// ValidateBasic does a stateless sanity check on the provided data.
+func (m MsgCreate) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Curator); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrap("malformed curator address " + err.Error())
+	}
+	if m.Name == "" || len(m.Name) > 100 {
```

Comment:
Even 100 is a bit long

Reply:
changed to 32, let's update later once we will have consensus about name and denom.